### DISCUSSION TITLE
Make dependency on AUX1 optional at Compile Time

### DISCUSSION THREAD

**Discussion by MartinHugh** [ExpressLRS/ExpressLRS#1468] [Ideas]
The Arm state influences the way the firmware works by affecting things like dynamic range and ground features. The Arm state is tied to AUX1.

Given the need to expedite channel information for some use cases, the use of AUX1 as the unequivocal Arming channel makes some sense.

However, this constraint makes two assumptions :

1) that FC Arming is needed in the first place. Many fixed/flying wing aircraft fly without Flight Controllers and do not need Arming state represented, and even if they do need it representing, it is often handled within the TX only, and not even transmitted.  Also some aircraft dont even have motors, like gliders.

2) that AUX1 already does not already have a dependency on something else required to fly.  ZOHD Kopilot is a RTH/stabiliser component whose flight mode channel (when passed by SBUS) is AUX1. This is not configurable, and use cases where FC arming is not needed, and a Kopilot would be used means that Express LRS cannot be used.

Allow features which are dependant on the state of AUX1 at runtime, to be overridden by compile-time options, so that AUX1 can be freed up for other use when not needed.

**Comment by robhaswell** [comment]
This seems related to #1486.
